### PR TITLE
Feature/flow images from dataframe

### PR DESCRIPTION
**Pull request by dfalbel**
This will add `flow_images_from_dataframe` function. Fixing #658. 
Needs more testing before merging.

### CODE REVIEW

File: tests/testthat/test-preprocessing.R [hunk 1]
```diff
@@ -75,5 +75,34 @@ test_succeeds("images arrays can be resized", {
   }
 })
 
+test_succeeds("flow images from dataframe works", {
+  
+  if (!reticulate::py_module_available("pandas"))
```
Comment (jjallaire) [inline tests/testthat/test-preprocessing.R]: If the function itself requires pandas we should note this in the documentation (and also do a check during the call to print a clear error message to that effect). Note that pandas will generally not be installed in TF environments installed via `install_tensorflow()` without special extra effort.

Comment (dfalbel) [reply tests/testthat/test-preprocessing.R]: Thanks JJ! I added a note to the docs here: https://github.com/rstudio/keras/pull/725/files#diff-151c30ec9c406dc7d593b4eeca5113a3R890 and I'll catch the error to print a clear message here: https://github.com/rstudio/keras/pull/725/files#diff-151c30ec9c406dc7d593b4eeca5113a3R910

File: R/preprocessing.R [hunk 2]
```diff
@@ -843,6 +843,100 @@ flow_images_from_directory <- function(
...
+  dataframe, directory = NULL, x_col = "filename", y_col = "class",
+  generator = image_data_generator(), target_size = c(256,256), 
+  color_mode = "rgb", classes = NULL, class_mode = "categorical", 
+  batch_size = 32, shuffle = TRUE, seed = NULL, save_to_dir = NULL, 
+  save_prefix = "", save_format = "png", subset = NULL, 
+  interpolation = "nearest", drop_duplicates = TRUE) {
+  
+  if (!reticulate::py_module_available("pandas"))
+    stop("Pandas (python module) must be installed in the same environment as Keras.", 
+         '. Install it using reticulate::py_install("pandas", envname = "r-tensorflow").')
```
Comment (jjallaire) [inline R/preprocessing.R]: This does leave open whether they used virtualenv or condaenv to install (the code above will default to virtualenv). 

They could also do this: `install_tensorflow(extra_packages = "pandas")`. Not sure if that's a better thing to document or not though (will leave that to your discretion!)

Comment (dfalbel) [reply R/preprocessing.R]: Ok, I thnk it's better now: https://github.com/rstudio/keras/pull/725/commits/643bca3038d77b8525fefe4bf1d0ad7f5f2037a0#diff-151c30ec9c406dc7d593b4eeca5113a3R914

**Comment by jjallaire** [comment]
Okay, that looks great!

**Pull request merged**